realm/realm-library/src/main/java/io/realm/RealmBaseSchema.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.Map;
+import java.util.Set;
+
+import io.realm.internal.ColumnIndices;
+import io.realm.internal.ColumnInfo;
+import io.realm.internal.RealmProxyMediator;
+import io.realm.internal.Table;
+
+
+/**
+ * Class for interacting with the Realm schema using a dynamic API. This makes it possible
+ * to add, delete and change the classes in the Realm.
+ * <p>
+ * All changes must happen inside a write transaction for the particular Realm.
+ *
+ * @see RealmMigration
+ */
+public abstract class RealmBaseSchema {
```

Comment:
The relationship between this change and the backlinks code is ephemeral but present.  The information about the backlinked fields, necessary for querying, has to live in some schema-like-object.  With the current organization, two implementations separated by if-then-elses, it was difficult for me (still trying to understand this thing) to figure out which of the two schema-like objects to choose: the `RealmObjectSchema`, or the `ColumnInfo` object.  Once I'd done the work to figure it out I thought I'd propose this code change, which will make this sort of change (which seems likely to happen a lot, as we move to the ObjectStore) much easier to do.